osrf/vrx
# Add a publisher rate to the thrust plugin

Partially addresses issue #340. 

This patch removes the warning messages:

```
 at line          at line 278278 in  in /tmp/binarydeb/ros-noetic-tf2-0.7.5/src/buffer_core.cpp/tmp/binarydeb/ros-noetic-tf2-0.7.5/src/buffer_core.cpp

Warning: Warning: TF_REPEATED_DATA ignoring data with redundant timestamp for frame wamv/lateral_propeller_link at time 471.232000 according to authority unknown_publisherTF_REPEATED_DATA ignoring data with redundant timestamp for frame wamv/lateral_propeller_link at time 471.232000 according to authority unknown_publisher
```

Here's my explanation about the warning and why it's fixed in the PR:

My understanding of this warning is that during a single state estimation update, the `ekf_localization_node` receives more than one `tf transform` from the same link. I believe the reason is because we were publishing the thuster joint states at  high frequency (1Khz). The `robot_state_publisher` is subscribed to `/wamv/joint_states` and publishes `/tf` data. `ekf_localization_node` is subscribed to `/tf` to receive all the transforms.

Reducing the joint publication rate from 1KHz to 100Hz makes the `ekf_localization_node` happy.

How to test it:

1. Launch simulation as usual:

```
roslaunch vrx_gazebo vrx.launch extra_gazebo_args:="--verbose"
```

2. Launch the robot_localization node:
```
roslaunch wamv_gazebo localization_example.launch
```

Verify that the `TF_REPEATED_DATA` warnings don't show up anymore.

3. Run rviz and verify that localization still looks good when you move the robot:
```
roslaunch wamv_gazebo rviz_vrx.launch
```

Note: The outstanding warning shown below is expected to be displayed. It's not tackled in this pull request.
```
[ WARN] [1634842636.098529156, 110.937000000]: Transform from wamv/gps_wamv_link to wamv/base_link was unavailable for the time requested. Using latest instead.

[ WARN] [1634842637.847973935, 112.667000000]: Transform from wamv/imu_wamv_link to wamv/base_link was unavailable for the time requested. Using latest instead.

```

cc @tfoote
